### DISCUSSION TITLE
About override new object (不能通过内存的方式创建对象，对于重写了new的class)

### DISCUSSION THREAD

**Discussion by 061999** [jrouwe/JoltPhysics#986] [Q&A]
![image](https://github.com/jrouwe/JoltPhysics/assets/88932040/9d87dae1-bb28-4819-bb54-c3a506f6f852)
为什么不能通过这种方式创建对象（Why can't objects be created in this way）

**Comment by 061999** [comment]
![image](https://github.com/jrouwe/JoltPhysics/assets/88932040/f38c7b86-df6a-4759-aabd-7f10bc5b6571)

**Comment by jrouwe** [comment]
Because operator `new` for `PhysicsSystem` is overridden, you need to do:

```
auto ptr = malloc(sizeof(JPH::PhysicsSystem));
auto phys = ::new (ptr) JPH::PhysicsSystem();
```
